specification/app/resource-manager/Microsoft.App/preview/2023-08-01-preview/Patches.json

```diff
@@ -0,0 +1,542 @@
+          "type": "string",
+          "format": "date-time",
+          "description": "The UTC timestamp that describes the latest detection was done."
+        },
+        "detectionStatus": {
+          "$ref": "#/definitions/DetectionStatus",
+          "description": "The status of the patch detection.",
+          "readOnly": true
+        },
+        "newImageName": {
+          "type": "string",
+          "description": "The name of the new image created by the patch.",
+          "x-nullable": true
+        },
+        "newLayer": {
+          "type": "object",
+          "properties": {
+            "name": {
+              "type": "string",
+              "description": "The details of the new layer for the target image."
+            },
+            "frameworkAndVersion": {
+              "type": "string",
+              "description": "The framework and its version in the new run image for the target image."
+            },
+            "osAndVersion": {
+              "type": "string",
+              "description": "The OS name and its version in the new run image for the target image."
+            }
+          },
+          "description": "The details of new run image for the target image.",
+          "x-nullable": true
+        },
+        "oldLayer": {
+          "type": "object",
+          "properties": {
+            "name": {
+              "type": "string",
+              "description": "The details of the old layer for the target image."
+            },
+            "frameworkAndVersion": {
+              "type": "string",
+              "description": "The framework and its version in the old run image for the target image."
+            },
+            "osAndVersion": {
+              "type": "string",
+              "description": "The OS name and its version in the old run image for the target image."
+            }
+          },
+          "description": "The details of old run image for the target image.",
+          "x-nullable": true
+        },
+        "patchType": {
+          "type": "string",
+          "description": "The type for the patch.",
+          "enum": [
+            "FrameworkSecurity",
+            "OSSecurity",
+            "FrameworkAndOS",
+            "Other"
+          ],
+          "x-nullable": true,
+          "x-ms-enum": {
+            "name": "PatchType",
+            "modelAsString": true
+          }
+        }
+      },
+      "readOnly": true
+    },
+    "PatchProperties": {
+      "type": "object",
+      "properties": {
+        "targetEnvironmentId": {
+          "format": "arm-id",
+          "description": "The id of the target environment for the patch."
+        },
+        "targetContainerAppId": {
+          "format": "arm-id",
+          "description": "The id of the target container app for the patch."
+        },
+        "targetRevisionId": {
+          "format": "arm-id",
+          "description": "The id of the target revision for the patch."
+        },
```

Comment:
I think you need type string in addition to format arm-id

          "type": "string",